python/lsst/meas/algorithms/loadReferenceObjects.py

```diff
@@ -43,6 +43,87 @@
 from lsst.daf.base import PropertyList
 
 
+def isOldFluxField(name, units):
+    """Return True if this name/units combination corresponds to an
+    "old-style" reference catalog flux field.
+    """
+    unitsCheck = units != 'nJy'  # (units == 'Jy' or units == '' or units == '?')
+    isFlux = name.endswith('_flux')
+    isFluxSigma = name.endswith('_fluxSigma')
+    isFluxErr = name.endswith('_fluxErr')
+    return (isFlux or isFluxSigma or isFluxErr) and unitsCheck
+
+
+def hasNanojanskyFluxUnits(schema):
+    """Return True if the units of all flux and fluxErr are correct (nJy).
+    """
+    for field in schema:
+        if isOldFluxField(field.field.getName(), field.field.getUnits()):
+            return False
+    return True
+
+
+def convertToNanojansky(catalog, log, doConvert=True):
+    """Convert fluxes in a catalog from jansky to nanojansky.
+
+    Parameters
+    ----------
+    catalog : `lsst.afw.table.SimpleCatalog`
+        The catalog to convert.
+    log : `lsst.log.Log`
+        Log to send messages to.
+    doConvert : `bool`, optional
+        Return a converted catalog, or just identify the fields that need to be converted?
+        This supports the "write=False" mode of `bin/convert_to_nJy.py`.
+
+    Returns
+    -------
+    catalog : `lsst.afw.table.SimpleCatalog` or None
+        The converted catalog, or None if ``doConvert`` is False.
+
+    Notes
+    -----
+    Support for old units in reference catalogs will be removed after 18.0.
```

Comment:
I think this may not happen until after v19, right?  Maybe it's safer to say it will be removed after the release of late calendar year 2019.